### PR TITLE
:bug: Remove the conditional on ipv6

### PR DESCRIPTION
**Pull request by ekarlso**
Signed-off-by: ekarlso <endre.karlson@xait.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by zhanggbj — COMMENTED** [review]
Thanks for the PR!
Just one nit, as we want to make Gateway optional for both IPv4 and IPv6, could you please also update the comment in L55 to make it consistent. Thanks!

**Comment by zhanggbj** [comment]
Hi @ekarlso, please also sign the CLA, you could follow the guide above :-)

**Comment by chrischdi** [comment]
This is for #2396

File: pkg/services/govmomi/ipam/parse.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func parseAddressWithPrefix(ipamAddress *ipamv1.IPAddress) (netip.Prefix, error)
 // family must match the other addresses of the same family. IPv4 Gateways are
 // required, but IPv6 gateways are not.
```
Comment (chrischdi) [inline pkg/services/govmomi/ipam/parse.go:55]: We should also update the godoc here

```suggestion
// family must match the other addresses of the same family. A gateway address
// is optional. If it is not set this function returns `nil, nil`.
```

**Comment by killianmuldoon** [comment]
/retitle :bug: Remove the conditional on ipv6

**Comment by chrischdi** [comment]
This PR also needs to fix the relevant tests :-)

**Review by killianmuldoon — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by killianmuldoon** [comment]
Should we backport this to 1.8 / 1.7?

**Comment by chrischdi** [comment]
> Should we backport this to 1.8 / 1.7?

I was thinking about that too. Would make sense to me to do so. It reduces the restrictions on the field so it should be okay?!

**Comment by killianmuldoon** [comment]
Yeah - absolutely fine with backporting this as a bug.

**Comment by killianmuldoon** [comment]
@chrischdi  - looks like we might have an issue with the cherrypick bot :thinking: 

I'll open an issue over at test infra

**Pull request merged**